pallets/pallet-dip-receiver/src/traits.rs

```diff
@@ -0,0 +1,52 @@
+// KILT Blockchain – https://botlabs.org
+// Copyright (C) 2019-2023 BOTLabs GmbH
+
+// The KILT Blockchain is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The KILT Blockchain is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+// If you feel like getting in touch with us, you can do so at info@botlabs.org
+
+use dip_support::VersionedIdentityProof;
+use sp_std::marker::PhantomData;
+
+pub trait IdentityProofVerifier {
+	type ProofDigest;
+	type LeafKey;
+	type LeafValue;
+	type VerificationResult;
+	type Error;
+
+	fn verify_proof_against_digest(
+		proof: VersionedIdentityProof<Self::LeafKey, Self::LeafValue>,
+		digest: Self::ProofDigest,
+	) -> Result<Self::VerificationResult, Self::Error>;
+}
+
+// Always returns success.
+pub struct SuccessfulProofVerifier<ProofDigest, LeafKey, LeafValue>(PhantomData<(ProofDigest, LeafKey, LeafValue)>);
```

Comment:
this should probably only be used in tests and benchmarks?

Reply:
in some pallets with have a mock feature for that

Reply:
It is used in the actual runtime template, so we can't use any features. It will be deprecated with the new merkle proof PR, but here it's used here: https://github.com/KILTprotocol/kilt-node/pull/489/files#diff-6c40e72378ea467c86621567df431cc2dabb74d5b76ef2a25fa43a16cf568e00R32